src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,222 @@
+    // Calculate cumulative widths of next siblings to get the the total offset of each of the cells
+    lastColumnsOffsets = lastColumnsOffsets
+      .map((elem, index) => lastColumnsOffsets.slice(0, index + 1).reduce((a, b) => a + b))
+      .reverse();
+
+    setCellOffsets({ first: [0, ...firstColumnsOffsets], last: [...lastColumnsOffsets, 0] });
+  }, [tableCellRefs]);
+
+  useLayoutEffect(() => {
+    // Request animation frame to make sure effect runs after the browser's automatic table layout algorithm
+    const frameId = requestAnimationFrame(() => {
+      updateCellOffsets();
+    });
+
+    return () => {
+      cancelAnimationFrame(frameId);
+    };
+  }, [updateCellOffsets]);
+
+  return { cellOffsets, updateCellOffsets };
+};
+
+interface GetStickyClassNamesProps {
+  styles: { [key: string]: string };
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  isStuckToTheLeft: boolean;
+  isStuckToTheRight: boolean;
+}
+
+export function getStickyClassNames({
+  styles,
+  isSticky,
+  isLastStickyLeft,
+  isLastStickyRight,
+  isStuckToTheLeft,
+  isStuckToTheRight,
+}: GetStickyClassNamesProps) {
+  return clsx({
+    [styles['sticky-cell']]: isSticky,
+    [styles['sticky-cell-last-left']]: isStuckToTheLeft && isLastStickyLeft,
+    [styles['sticky-cell-last-right']]: isStuckToTheRight && isLastStickyRight,
+  });
+}
+
+export const useStickyColumns = ({
+  containerWidth,
+  hasSelection,
+  isWrapperScrollable,
+  stickyColumns,
+  tablePadding,
+  visibleColumnsLength,
+  wrapperRefObject,
+}: StickyColumnParams) => {
+  // Check if there are any sticky columns
+  const noStickyColumns = !stickyColumns?.first && !stickyColumns?.last;
+  const columnsLengthWithSelection = visibleColumnsLength + (hasSelection ? 1 : 0);
+
+  const [shouldDisable, setShouldDisable] = useState<boolean>(noStickyColumns);
+  const [tableCellRefs, setTableCellRefs] = useState<Array<React.RefObject<HTMLTableCellElement>>>([]);
+  const { cellOffsets, updateCellOffsets } = useCellOffsets(tableCellRefs);
+
+  const { first = 0, last = 0 } = stickyColumns || {};
+  // Calculate the indexes of the last left and right sticky columns, taking into account the selection column
+  const lastLeftStickyColumnIndex = first + (hasSelection ? 1 : 0);
+  const lastRightStickyColumnIndex = columnsLengthWithSelection - 1 - last;
+
+  // Get the width of the first and last sticky columns using the `cellOffsets` state, or use 0 if it's not available
+  const firstStickyColumnsWidth = cellOffsets.first[lastLeftStickyColumnIndex] ?? 0;
+  const lastStickyColumnsWidth = cellOffsets.last[lastRightStickyColumnIndex] ?? 0;
+
+  // Calculate the sum of all sticky columns' widths
+  const totalStickySpace = firstStickyColumnsWidth + lastStickyColumnsWidth;
+
+  useEffect(() => {
+    // If scroll width changes we recalculate the cell offsets
+    updateCellOffsets();
+  }, [updateCellOffsets, wrapperRefObject?.current?.scrollWidth]);
```

Comment:
The second argument change does not cause a re-render. I would challenge the need of having that in the dependencies array as it would only works if there was an associated state update and therefore unreliable.